src/index.ts

```diff
@@ -197,6 +214,20 @@ export class PubSubPeerDiscovery extends TypedEventEmitter<PeerDiscoveryEvents>
       return
     }
 
+    // Broadcast on Subscribe from other peers
+    if (this.broadcastOnSubscribe) {
+      pubsub.addEventListener('subscription-change', subChangeEvt => {
+        // Check if the PubSub peer cares about PubSub Peer Discovery
+        const discoverySubs = subChangeEvt.detail.subscriptions.filter(sub => this.topics.includes(sub.topic));
+
+        // The Peer is interested in PubSub Peer Discovery -> broadcast
+        if (discoverySubs.length > 0) {
```

Comment:
No need to check every subscription, just the first that matches:

```suggestion
        const subscribedToDiscovery = subChangeEvt.detail.subscriptions.some(sub => this.topics.includes(sub.topic));

        // The Peer is interested in PubSub Peer Discovery -> broadcast
        if (subscribedToDiscovery) {
```